### ISSUE TITLE
Segmentation Fault 

### ISSUE THREAD

**Issue by trungmanhhuynh** [NVIDIA/flownet2-pytorch#186]
I got this segmentation fault when running the inference script. Please help !!!

python main.py --inference --model FlowNet2 --save_flow --save ./output --inference_dataset ImagesFromFolder --inference_dataset_root /home/manhh/github/datasets/MPI-Sintel/MPI-Sintel-testing/test/clean/market_1/ --resume checkpoints/FlowNet2_checkpoint.pth.tar 
Parsing Arguments
  [2.810s] batch_size: 8
  [2.810s] crop_size: [256, 256]
  [2.810s] fp16: False
  [2.810s] fp16_scale: 1024.0
  [2.810s] gradient_clip: None
  [2.810s] inference: True
  [2.810s] inference_batch_size: 1
  [2.810s] inference_dataset: ImagesFromFolder
  [2.810s] inference_dataset_iext: png
  [2.810s] inference_dataset_replicates: 1
  [2.810s] inference_dataset_root: /home/manhh/github/datasets/MPI-Sintel/MPI-Sintel-testing/test/clean/market_1/
  [2.810s] inference_n_batches: -1
  [2.810s] inference_size: [-1, -1]
  [2.810s] log_frequency: 1
  [2.810s] loss: L1Loss
  [2.810s] model: FlowNet2
  [2.810s] model_batchNorm: False
  [2.810s] model_div_flow: 20.0
  [2.810s] name: run
  [2.810s] no_cuda: False
  [2.810s] number_gpus: 4
  [2.810s] number_workers: 8
  [2.810s] optimizer: Adam
  [2.810s] optimizer_amsgrad: False
  [2.810s] optimizer_betas: (0.9, 0.999)
  [2.810s] optimizer_eps: 1e-08
  [2.810s] optimizer_lr: 0.001
  [2.810s] optimizer_weight_decay: 0
  [2.810s] render_validation: False
  [2.810s] resume: checkpoints/FlowNet2_checkpoint.pth.tar
  [2.810s] rgb_max: 255.0
  [2.810s] save: ./output
  [2.810s] save_flow: True
  [2.810s] schedule_lr_fraction: 10
  [2.810s] schedule_lr_frequency: 0
  [2.810s] seed: 1
  [2.810s] skip_training: False
  [2.810s] skip_validation: False
  [2.810s] start_epoch: 1
  [2.810s] total_epochs: 10000
  [2.810s] train_n_batches: -1
  [2.810s] training_dataset: MpiSintelFinal
  [2.810s] training_dataset_replicates: 1
  [2.810s] training_dataset_root: ./MPI-Sintel/flow/training
  [2.810s] validation_dataset: MpiSintelClean
  [2.810s] validation_dataset_replicates: 1
  [2.810s] validation_dataset_root: ./MPI-Sintel/flow/training
  [2.810s] validation_frequency: 5
  [2.810s] validation_n_batches: -1
  [2.810s] Operation finished

Source Code
  Current Git Hash: b'71034046166735a79a5b82df78de72d806e82842'

Initializing Datasets
  [0.040s] Inference Dataset: ImagesFromFolder
  [0.090s] Inference Input: [3, 2, 384, 1024]
  [0.140s] Inference Targets: [3, 2, 384, 1024]
  [0.140s] Operation finished

Building FlowNet2 model
  [4.010s] Effective Batch Size: 32
  [4.010s] Number of parameters: 162518834
  [4.010s] Initializing CUDA
  [7.210s] Parallelizing
  [7.210s] Loading checkpoint 'checkpoints/FlowNet2_checkpoint.pth.tar'
  [7.560s] Loaded checkpoint 'checkpoints/FlowNet2_checkpoint.pth.tar' (at epoch 0)
  [7.560s] Initializing save directory: ./output
  [7.570s] Operation finished

Initializing Adam Optimizer
  [0.000s] amsgrad = False (<class 'bool'>)
  [0.000s] weight_decay = 0 (<class 'int'>)
  [0.000s] eps = 1e-08 (<class 'float'>)
  [0.000s] betas = (0.9, 0.999) (<class 'tuple'>)
  [0.000s] lr = 0.001 (<class 'float'>)
  [0.000s] Operation finished

Overall Progress:   0%|                                                       | 0/1 [00:00<?, ?it/s]**Segmentation fault (core dumped)**

I am using: 
pytorch 1.0.0. 
python 3.6.8
cuda 9.0

Anyone had this resolved, please help! Thanks a lot.

**Comment by trungmanhhuynh** [comment]
I've had this problem resolved, I use gcc5.4.0, centos 7.0, pytorh 1.0.0 and other conda envs as below: 
Hi @leeeeeeo, he below conda enviroment works: 
'''
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main  
blas                      1.0                         mkl  
ca-certificates           2019.10.16                    0  
certifi                   2019.9.11                py36_0  
cffi                      1.13.2           py36h2e261b9_0  
colorama                  0.4.1                     <pip>
cudatoolkit               9.0                  h13b8566_0  
cudnn                     7.6.4                 cuda9.0_0  
cycler                    0.10.0                    <pip>
decorator                 4.4.1                     <pip>
imageio                   2.6.1                     <pip>
intel-openmp              2019.4                      243  
kiwisolver                1.1.0                     <pip>
libffi                    3.2.1                hd88cf55_4  
libgcc-ng                 9.1.0                hdf63c60_0  
libgfortran-ng            7.3.0                hdf63c60_0  
libstdcxx-ng              9.1.0                hdf63c60_0  
matplotlib                3.1.1                     <pip>
mkl                       2018.0.3                      1  
mkl_fft                   1.0.6            py36h7dd41cf_0  
mkl_random                1.0.1            py36h4414c95_1  
nccl                      1.3.5                 cuda9.0_0  
networkx                  2.4                       <pip>
ninja                     1.9.0            py36hfd86e86_0  
numpy                     1.15.4           py36h1d66e8a_0  
numpy-base                1.15.4           py36h81de0dd_0  
openssl                   1.0.2t               h7b6447c_1  
Pillow                    6.2.1                     <pip>
pip                       19.3.1                   py36_0  
protobuf                  3.10.0                    <pip>
pycparser                 2.19                       py_0  
pyparsing                 2.4.5                     <pip>
python                    3.6.1                         2  
python-dateutil           2.8.1                     <pip>
pytorch                   1.0.0           py3.6_cuda9.0.176_cudnn7.4.1_1    pytorch
pytz                      2019.3                    <pip>
PyWavelets                1.1.1                     <pip>
readline                  6.2                           2  
scikit-image              0.16.2                    <pip>
scipy                     1.1.0                     <pip>
setproctitle              1.1.10                    <pip>
setuptools                41.6.0                   py36_0  
six                       1.13.0                    <pip>
sqlite                    3.13.0                        0  
tbb                       2019.8               hfd86e86_0  
tbb4py                    2019.8           py36hfd86e86_0  
tensorboardX              1.9                       <pip>
tk                        8.5.18                        0  
tqdm                      4.38.0                    <pip>
wheel                     0.33.6                   py36_0  
xz                        5.2.4                h14c3975_4  
zlib                      1.2.11               h7b6447c_3 
'''

**Comment by wxz0918** [comment]
how did you solve this problem

**Comment by wxz0918** [comment]
@trungmanhhuynh

**Comment by hsyi** [comment]
same problem, how to solve it

**Comment by trungmanhhuynh** [comment]
Hi, 
As I remember, the problem caused by the inconsistent cuda versions among packages that relies on cuda. 
So make sure you have 
cudatoolkit 9.0 h13b8566_0
cudnn 7.6.4 cuda9.0_0

**Comment by javlonbek97** [comment]
How about if cuda 10.1 and torch 1.5.1 are used?
